godot-core/src/registry/signal/typed_signal.rs

```diff
@@ -189,6 +198,48 @@ impl<C: WithSignals, Ps: InParamTuple + 'static> TypedSignal<'_, C, Ps> {
     ///
     /// - To connect to methods on other objects, use [`connect_other()`][Self::connect_other].
     /// - If you need [`connect flags`](ConnectFlags) or cross-thread signals, use [`builder()`][Self::builder].
+    ///
+    /// # Example
+    /// ```no_run
+    /// # use godot::prelude::*;
+    /// # #[derive(GodotClass)]
+    /// # #[class(init, base=Node)]
+    /// # pub struct Player {
+    /// #     health_ui: OnEditor<Gd<HealthUI>>,
+    /// #     base: Base<Node>,
+    /// # }
+    /// # #[derive(GodotClass)]
+    /// # #[class(init, base=Node2D)]
+    /// # pub struct HealthUI {
+    /// #     base: Base<Node2D>,
+    /// # }
+    /// # #[godot_api]
+    /// # impl Player {
+    /// #     #[signal]
+    /// #     fn health_changed(health: i32);
+    /// # }
+    ///  impl Player {
+    ///      fn change_health_anim(&mut self, health: i32) {
+    ///         // Change animation based on health
+    ///      }
+    ///  }
+    ///  impl HealthUI {
+    ///      fn on_health_changed(&mut self, health: i32) {
+    ///         // Change healthbar UI based on health
+    ///      }
+    ///  }
+    /// #[godot_api]
+    /// impl INode for Player {
+    ///     fn ready(&mut self) {
+    ///         self.signals() // Connect to self
+    ///             .health_changed()
+    ///             .connect_self(Self::change_health_anim);
+    ///         self.signals() // Connect to other object (health_ui is OnEditor<Gd<HealthUI>>)
+    ///             .health_changed()
+    ///             .connect_self(|s, amount| s.health_ui.bind_mut().on_health_changed(amount));
+    ///     }
```

Comment:
This is confusing:

> Connect to other object

That's not true, you still connect to _this_ object. You just manually delegate in to the other UI. Which raises more questions than it answers:
- Why not connect a signal directly to the UI?
- If not, why not address this in the same handler as `change_health_anim`?
- How are these two different from a connect perspective?

Reply:
I think this example is more complex than it needs to be. If the point is to demonstrate that one can use both function references and closures, then I'd use _the same_ connection, with two different syntaxes. That's much clearer and immediately demonstrates the equivalence.

Reply:
Yeah, you can't use `connect_other` on `self`, that's why `connect_self` exists.

What I meant is: show the exact same `connect_self` call twice:
- with a function reference
- with a closure

This makes clear that it's possible to use different syntaxes.